lib/jekyll/tags/include.rb

```diff
@@ -59,7 +63,7 @@ def parse_params(context)
       end
 
       def validate_file_name(file)
-        if file !~ %r!^[a-zA-Z0-9_/\.-]+$! || file =~ %r!\./! || file =~ %r!/\.!
+        if file =~ INVALID_SEQUENCES || file !~ VALID_FILENAME_CHARS
```

Comment:
The other two (below) are obvious wins. This one needs a closer look. I am worried about unintentionally changing behavior. It is not obvious to me that `INVALID_SEQUENCES` is the complement of `%r!^[a-zA-Z0-9_/\.-]+$!`, or what this is all about. Since the existing regexes are kind of cryptic, I might need to `git blame` to figure out what the original intent was.

Reply:
This line is nine years old.
https://github.com/jekyll/jekyll/commit/8a2a42ba7142f705bb2ef31ec204d0e3c3beef75#diff-a22f622dd552a293d1933aae37f4fb5bR10

Reply:
> It is not obvious to me that `INVALID_SEQUENCES` is the complement of `%r!^[a-zA-Z0-9_/\.-]+$!`

I switched the calls here.. (as mentioned in the PR opener..)
`INVALID_SEQUENCES = %r![\./][\./]+!`
`VALID_FILENAME_CHARS = %r!^[\w/\.-]+$!`